plugin/evm/vm_test.go

```diff
@@ -78,25 +78,57 @@ var (
+
+	genesisJSON = func(cfg *params.ChainConfig) string {
+		g := new(core.Genesis)
+		g.Difficulty = big.NewInt(0)
+		g.GasLimit = 0x5f5e100
+
+		// Use chainId: 43111, so that it does not overlap with any Avalanche ChainIDs, which may have their
+		// config overridden in vm.Initialize.
+		cpy := *cfg
+		cpy.ChainID = big.NewInt(43111)
+		g.Config = &cpy
+
+		allocStr := `{"0100000000000000000000000000000000000000":{"code":"0x7300000000000000000000000000000000000000003014608060405260043610603d5760003560e01c80631e010439146042578063b6510bb314606e575b600080fd5b605c60048036036020811015605657600080fd5b503560b1565b60408051918252519081900360200190f35b818015607957600080fd5b5060af60048036036080811015608e57600080fd5b506001600160a01b03813516906020810135906040810135906060013560b6565b005b30cd90565b836001600160a01b031681836108fc8690811502906040516000604051808303818888878c8acf9550505050505015801560f4573d6000803e3d6000fd5b505050505056fea26469706673582212201eebce970fe3f5cb96bf8ac6ba5f5c133fc2908ae3dcd51082cfee8f583429d064736f6c634300060a0033","balance":"0x0"}}`
+		json.Unmarshal([]byte(allocStr), &g.Alloc)
+		// After Durango, an additional account is funded in tests to use
+		// with warp messages.
+		if cfg.IsDurango(0) {
+			addr := common.HexToAddress("0x99b9DEA54C48Dfea6aA9A4Ca4623633EE04ddbB5")
+			balance := new(big.Int).Mul(big.NewInt(params.Ether), big.NewInt(10))
+			g.Alloc[addr] = core.GenesisAccount{Balance: balance}
+		}
+
+		b, err := json.Marshal(g)
+		if err != nil {
+			panic(err)
+		}
+		return string(b)
+	}
+
+	activateCancun = func(cfg *params.ChainConfig) *params.ChainConfig {
+		cpy := *cfg
+		cpy.CancunTime = utils.NewUint64(0)
+		return &cpy
+	}
+
+	genesisJSONApricotPhase0     = genesisJSON(params.TestLaunchConfig)
+	genesisJSONApricotPhase1     = genesisJSON(params.TestApricotPhase1Config)
+	genesisJSONApricotPhase2     = genesisJSON(params.TestApricotPhase2Config)
+	genesisJSONApricotPhase3     = genesisJSON(params.TestApricotPhase3Config)
+	genesisJSONApricotPhase4     = genesisJSON(params.TestApricotPhase4Config)
+	genesisJSONApricotPhase5     = genesisJSON(params.TestApricotPhase5Config)
+	genesisJSONApricotPhasePre6  = genesisJSON(params.TestApricotPhasePre6Config)
+	genesisJSONApricotPhase6     = genesisJSON(params.TestApricotPhase6Config)
+	genesisJSONApricotPhasePost6 = genesisJSON(params.TestApricotPhasePost6Config)
+	genesisJSONBanff             = genesisJSON(params.TestBanffChainConfig)
+	genesisJSONCortina           = genesisJSON(params.TestCortinaChainConfig)
+	genesisJSONDurango           = genesisJSON(params.TestDurangoChainConfig)
+	genesisJSONEUpgrade          = genesisJSON(params.TestEUpgradeChainConfig)
+	genesisJSONLatest            = genesisJSONEUpgrade
+
+	genesisJSONCancun = genesisJSON(activateCancun(params.TestChainConfig))
```

Comment:
note this is not a behavior change, the cancun activation with E-Upgrade will be a separate PR